src/main/java/org/jenkinsci/plugins/build_token_root/BuildRootAction.java

```diff
@@ -83,8 +86,12 @@ public void doBuild(StaplerRequest req, StaplerResponse rsp, @QueryParameter Str
             LOGGER.fine("wrong kind");
             throw HttpResponses.error(HttpServletResponse.SC_BAD_REQUEST, "Use /buildByToken/buildWithParameters for this job since it takes parameters");
         }
-        Jenkins.getInstance().getQueue().schedule(p, delay.getTime(), getBuildCause(req));
-        ok(rsp);
+        Queue.Item item = Jenkins.getInstance().getQueue().schedule(p, delay.getTime(), getBuildCause(req));
+        if (item != null) {
+            rsp.sendRedirect(SC_CREATED, req.getContextPath() + '/' + item.getUrl());
```

Comment:
Has this been tested - to redirect you would need a 301 or 302 response code - so `SC_CREATED` is incorrect or ignored.